tests/test_logsumexp.py

```diff
@@ -6,23 +9,115 @@
 
 
 @pytest.mark.memcheck
-@pytest.mark.parametrize("precision,atol,rtol", [(np.float64, 1e-8, 1e-8), (np.float32, 2e-5, 1e-5)])
-@pytest.mark.parametrize("seed", list(range(2)))
+@pytest.mark.parametrize("precision", [np.float64, np.float32])
+def test_segmented_cuda_logsumexp_validation(precision):
+    max_values_per_segment = 5
+    num_segments = 2
+    if precision == np.float32:
+        summer = custom_ops.SegmentedSumExp_f32(max_values_per_segment, num_segments)
+    else:
+        summer = custom_ops.SegmentedSumExp_f64(max_values_per_segment, num_segments)
+
+    with pytest.raises(RuntimeError, match="number of segments don't match"):
+        summer.logsumexp([[1.0]])
+
+    with pytest.raises(RuntimeError, match="empty array not allowed"):
+        summer.logsumexp(
+            [
+                [
+                    1.0,
+                ],
+                [],
+            ]
```

Comment:
nit: remove trailing `,` to format more compactly (unless intentional?)

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1239/commits/43a2967746b863a8efd0b6f614ebd3f7ae514b1d